core/src/main/java/org/apache/carbondata/core/util/TrashUtil.java

```diff
@@ -208,6 +205,14 @@ public static boolean isTrashRetentionTimeoutExceeded(long fileTimestamp) {
     return difference > retentionMilliSeconds;
   }
 
+  /**
+   * whether trash data outside of .Trash folder is time out
+   */
+  public static boolean isTrashDataTimeout(long fileTimestamp) {
+    return isTrashRetentionTimeoutExceeded(fileTimestamp) &&
```

Comment:
In the default case, when isTrashRetentionTimeoutExceeded  , isMaxQueryTimeoutExceeded will always be true. 
why not remove `queryMaxTimeout` and in query flow clean the metadata indexed trash based on `trashRetentionTimeout` ??

Reply:
isTrashRetentionTimeoutExceeded:  whether trash data inside of .Trash folder is time out
isTrashDataTimeout: whether trash data outside of .Trash folder is time out

TrashRetentionTimeout can be 0, but MaxQueryTimeout not.

Reply:
ok